xtask/src/build_docker_image.rs

```diff
@@ -107,17 +107,25 @@ pub fn build_docker_image(
         if gha && targets.len() > 1 {
             println!("::group::Build {target}");
         }
+        let docker_root = metadata.workspace_root.join("docker");
         let mut docker_build = Command::new(engine);
         docker_build.args(&["buildx", "build"]);
-        docker_build.current_dir(metadata.workspace_root.join("docker"));
+        docker_build.current_dir(&docker_root);
 
         if push {
             docker_build.arg("--push");
         } else {
             docker_build.arg("--load");
         }
 
-        let dockerfile = format!("Dockerfile.{target}");
+        let dockerfile_name = format!("Dockerfile.{target}");
+        let dockerfile_root = if docker_root.join(&dockerfile_name).exists() {
+            docker_root.clone()
+        } else {
+            docker_root.join("cross-toolchains").join("docker")
```

Comment:
this would make the `docker buildx build` say that a file at `cross-toolchains` does not exist. I think it's better to instead of doing this, to check if the file exists at `cross-toolchains`, and if it does not fallback to `docker/`.

Reply:
Done, however, the error message is still fairly opaque. Here's a sample where I replace `s390x-unknown-linux-gnu-cross` with `s390x-unknown-linux-gnu-crossx`:

```bash
+ "/home/ahuszagh/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo" "metadata" "--format-version=1" "--no-deps"                                                                         
+ "/usr/bin/docker" "buildx" "build" "--load" "--pull" "--cache-from" "type=registry,ref=ghcr.io/cross-rs/s390x-unknown-linux-gnu-crossx:main" "--tag" "ghcr.io/cross-rs/s390x-unknown-linux-gnu-cro
ssx:local" "-f" "/home/ahuszagh/git/cross/docker/Dockerfile.s390x-unknown-linux-gnu-crossx" "--progress" "auto" "."                                                                                 
[+] Building 0.2s (2/2) FINISHED                                                                                                                                                                    
 => [internal] load build definition from Dockerfile.s390x-unknown-linux-gnu-crossx                                                                                                            0.1s 
 => => transferring dockerfile: 2B                                                                                                                                                             0.0s 
 => [internal] load .dockerignore                                                                                                                                                              0.1s 
 => => transferring context: 2B                                                                                                                                                                0.0s 
error: failed to solve: rpc error: code = Unknown desc = failed to solve with frontend dockerfile.v0: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount1322602709/Dockerfile.s390x-
unknown-linux-gnu-crossx: no such file or directory                                                                                                                                                 
Error:                                                                                                                                                                                              
   0: encountered error(s)                                                                                                                                                                          
                                                                                                                                                                                                    
Location:                                                                                                                                                                                           
   xtask/src/build_docker_image.rs:245                                                                                                                                                              
                                                                                                                                                                                                    
Error:                                                                                                                                                                                              
   0: `"/usr/bin/docker" "buildx" "build" "--load" "--pull" "--cache-from" "type=registry,ref=ghcr.io/cross-rs/s390x-unknown-linux-gnu-crossx:main" "--tag" "ghcr.io/cross-rs/s390x-unknown-linux-gn
u-crossx:local" "-f" "/home/ahuszagh/git/cross/docker/Dockerfile.s390x-unknown-linux-gnu-crossx" "--progress" "auto" "."` failed with exit code: exit status: 1                                     
[ahuszagh@ahuszagh cross]$ git log 
```